src/main/docs/guide/languageSupport/kotlin/coroutines.adoc

```diff
@@ -0,0 +1,33 @@
+Kotlin coroutines allow you to create asynchronous applications with imperative style code. A Micronaut's controller
+action can be a `suspend` function:
+
+snippet::io.micronaut.docs.server.suspend.SuspendController[tags="suspend", indent="0", title="Controller suspend function example"]
+
+<1> The function is marked as `suspend`, though in reality it won't be suspended.
+
+NOTE: Functions declared in controllers that use `suspend` must be `open` as well.
```

Comment:
Another option would be to add io.micronaut.http.annotation.Controller to the all-open plugin config, just like it's done with io.micronaut.aop.Around. I've tested this locally and works fine.